docs/quickstarts/react-router.mdx

```diff
@@ -0,0 +1,270 @@
+      details = error.message
+      stack = error.stack
+    }
+
+    return (
+      <main className="pt-16 p-4 container mx-auto">
+        <h1>{message}</h1>
+        <p>{details}</p>
+        {stack && (
+          <pre className="w-full p-4 overflow-x-auto">
+            <code>{stack}</code>
+          </pre>
+        )}
+      </main>
+    )
+  }
+  ```
+
+  If you need to load in additional data, you can pass your loader directly to the `rootAuthLoader`.
+
+  ```tsx {{ filename: 'app/root.tsx' }}
+  // Imports
+
+  export async function loader(args: Route.LoaderArgs) {
+    return rootAuthLoader(args, ({ request }) => {
+      const { sessionId, userId, getToken } = request.auth
+      // Add logic to fetch data
+      return { yourData: 'here' }
+    })
+  }
+
+  // Rest of the root.tsx code
+  ```
+
+  ### Add `<ClerkProvider>`
+
+  The [`<ClerkProvider>`](/docs/components/clerk-provider) component wraps your app to provide active session and user context to Clerk's hooks and other components. The data from the root `loader` you added in the previous step is available as `loaderData` to the default App export inside `root.tsx`. Pass the `loaderData` to `<ClerkProvider>`.
+
+  ```tsx {{ filename: 'app/root.tsx' }}
+  // Imports
+  import { ClerkProvider } from '@clerk/react-router'
+
+  export default function App({ loaderData }: Route.ComponentProps) {
+    return (
+      <ClerkProvider loaderData={loaderData}>
+        <Outlet />
+      </ClerkProvider>
+    )
+  }
+
+  // Rest of the root.tsx code
+  ```
+
+  ### Add Clerk components
+
+  You can control which content signed-in and signed-out users can see with Clerk's [prebuilt control components](/docs/components/overview#what-are-control-components). Create a header using the following components:
+
+  - [`<SignedIn>`](/docs/components/control/signed-in): Children of this component can only be seen while **signed in**.
+  - [`<SignedOut>`](/docs/components/control/signed-out): Children of this component can only be seen while **signed out**.
+  - [`<UserButton />`](/docs/components/user/user-button): Shows the signed-in user's avatar. Selecting it opens a dropdown menu with account management options.
+  - [`<SignInButton />`](/docs/components/unstyled/sign-in-button): An unstyled component that links to the sign-in page. In this example, since no props or [environment variables](/docs/deployments/clerk-environment-variables) are set for the sign-in URL, this component links to the [Account Portal sign-in page](/docs/customization/account-portal/overview#sign-in).
+
+  Add them to the App component:
+
+  ```tsx {{ filename: 'app/root.tsx', mark: [2, [7, 17]] }}
+  // Imports
+  import { ClerkProvider, SignedIn, SignedOut, UserButton, SignInButton } from '@clerk/react-router'
+
+  export default function App({ loaderData }: Route.ComponentProps) {
+    return (
+      <ClerkProvider loaderData={loaderData}>
+        <header className="flex items-center justify-center py-8 px-4">
```

Comment:
We can use tailwind here to style stuff a bit but lemme know if we should remove it